### DISCUSSION TITLE
How to open a subfolder of a git repo without mounting its parent

### DISCUSSION THREAD

**Discussion by hellodword** [devcontainers/community#112] [Q&A]
Reproduce:

```sh
# init a new git repo and let's create some files
mkdir parent && cd parent
git init
touch file1

# create a subfolder and create a `.devcontainer` inside the subfolder
mkdir -p sub/.devcontainer
echo '{ "image": "mcr.microsoft.com/devcontainers/base:ubuntu-22.04" }' > sub/.devcontainer/devcontainer.json

# open the subfolder with VSCode
code sub
```

Get into the devcontainer, you will find not only the subfolder, but also the whole git repo is mounted into the container:

```sh
vscode ➜ /workspaces/parent/sub (master) $ ls -al /workspaces/parent/
total 0
drwxr-xr-x 4 vscode users 100 Feb 24 07:46 .
drwxr-xr-x 1 root   root   12 Feb 24 07:46 ..
-rw-r--r-- 1 vscode users   0 Feb 24 07:46 file1
drwxr-xr-x 7 vscode users 200 Feb 24 07:45 .git
drwxr-xr-x 3 vscode users  60 Feb 24 07:45 sub
```

What if I do not want to mount the whole git repo into the devcontainer?

A workaround is doing `git init` in the subfolder, but it's too hacky.

Any idea?

**Comment by samruddhikhandale** [comment]
Hi 👋 

Would you be interested in using the VS Code IDE? It will be easier to do with the IDE.

In Visual Studio Code, you can open a subfolder of a Git repository without mounting its parent by following these steps:

- Open Visual Studio Code.
- Click on File in the menu bar.
- Click on Open....
- Navigate to the subfolder you want to open.
- Click on Open.

This will open the subfolder in a new Visual Studio Code window. Note that Git operations will still consider the entire repository, not just the subfolder. This is because Git operates on the level of the entire repository, not individual folders within the repository.

**Reply by hellodword** [reply]
Thanks @samruddhikhandale 

I tried three ways:

1. `code parent/sub` -> `Devcontainers: Rebuild Without Cache and Reopen In Container`
2. `code parent` -> `Devcontainers: Rebuild Without Cache and Reopen In Container` -> `File - Open Folder ... - sub`
3. `code parent` -> `File - Open Folder ... - sub` -> `Devcontainers: Rebuild Without Cache and Reopen In Container`

All of them will mount the `parent` (or the path where the `.git` located) into the devcontainer
